x-pack/plugins/observability_onboarding/server/routes/elastic_agent/route.ts

```diff
@@ -28,8 +28,9 @@ const generateConfig = createObservabilityOnboardingServerRoute({
       esClient: coreStart.elasticsearch.client.asInternalUser as Client,
     });
 
-    const savedState = await findLatestObservabilityOnboardingState({
+    const savedState = await getObservabilityOnboardingState({
```

Comment:
One thing that we might want to do is to clean up the saved objects after they have served their purpose, although we expect the storage usage to be minimal.